app/models/invoice.rb

```diff
@@ -602,7 +606,7 @@ def buyer_field_label(name)
 
   # Returns years which have invoice
   def self.years
-    self.connection.select_values(selecting { sift(:year, period).as('year') }.reorder('year desc').to_sql).map(&:to_i)
+    self.connection.select_values(selecting { sift(:year, period).as('year') }.distinct.reorder('year desc').to_sql).map(&:to_i)
   end
```

Comment:
I think this issue could be resolved just ensuring this method doesn't return values previous to the account creation date or posterior to today. So avoiding the validation. But it's true that's a workaround, what do you think?

Reply:
Is this related to the issue with year validation? Or is unrelated?

Reply:
@akostadinov unrelated. I just noticed it when troubleshooting the issue.

Reply:
@jlledom This one is a different issue not related to the missing validation. There is a note in the PR description that explains why this is needed.
Basically, this method returns the years that are later going to be used for populating the select here: 
https://github.com/3scale/porta/blob/0e946cc325b60719fc3eff9820174b39da278765/app/views/finance/provider/invoices/index.html.erb#L28

Screenshot:
![invoices_years_dropdown](https://user-images.githubusercontent.com/1270328/195037855-6f374d28-e1b6-4a06-be07-8cd37fd69b7d.png)

The current method returns the array of years, the same size as invoices size. So, the same year will be repeated multiple times (a **LOT** of times), which doesn't make sense. We just want to know the list of **different** years.

Reply:
I know, but my comment is not about the `different` keyword here, I'm ok with that. What I mean is that this method could be modified to cut off all years before account creation date and after present year, so return only the valid range. That way the selector field would receive only the valid values and the whole issue would be solved with just that change.

Reply:
Yeah, I think we need the validation anyway.

Reply:
For me this is fine to stay as is without hiding something there even if it was improperly added somehow.